NPE when trying to download maven-metadata.xml

```
2016-03-16 14:41:34.326 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  org.commonjava.indy.util.AcceptInfo - Got raw ACCEPT header values:
  
2016-03-16 14:41:34.326 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.b.jaxrs.ContentAccessHandler - GET path: 'org/twitter4j/twitter4j/maven-metadata.xml' (RAW: '/group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml')
In store: 'group:builds-untested'
User accept header is: 'text/html'
Standard accept header for that is: 'text/html'
2016-03-16 14:41:34.326 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.a.d.1473720381$Proxy$_$$_WeldDecoratorProxy - DECORATED (getGroup: builds-untested)
2016-03-16 14:41:34.326 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.a.data.AutoProxCatalogManager - Autoprox is disabled.
2016-03-16 14:41:34.326 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.a.d.1473720381$Proxy$_$$_WeldDecoratorProxy - AutoProx decorator disabled; returning: Group [constituents=[hosted:build_keycloak-1-9-0-CR1-redhat-2-2_20160302.2106, hosted:build_jboss-modules-1-5-0_20160303.0619, hosted:build_20160303092153_kc-twitter_20160303.0922, hosted:build_20160303092153_kc-twitter_20160303.0924, hosted:build_20160303092153_kc-twitter_20160303.0925, hosted:build_jboss-modules-1-5-0_20160311.0431, hosted:build_jboss-modules-1-5-0_20160311.0434, hosted:build_jboss-modules-1-5-0-group-member_20160311.0520, hosted:build_jboss-modules-1-5-0_20160311.1521, hosted:build_jboss-servlet-spec-api-1-0-1_20160311.1525, hosted:build_org-liquibase-liquibase-parent-3-4-1_20160315.0805, hosted:build_com-google-zxing-zxing-parent-3-2-1_20160315.0807, hosted:Freemarker, hosted:build_com-google-zxing-zxing-parent-3-2-1_20160315.1355], getName()=builds-untested, getKey()=group:builds-untested]
2016-03-16 14:41:34.326 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.a.d.1473720381$Proxy$_$$_WeldDecoratorProxy - DECORATED (getGroup: builds-untested)
2016-03-16 14:41:34.326 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.a.data.AutoProxCatalogManager - Autoprox is disabled.
2016-03-16 14:41:34.326 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.a.d.1473720381$Proxy$_$$_WeldDecoratorProxy - AutoProx decorator disabled; returning: Group [constituents=[hosted:build_keycloak-1-9-0-CR1-redhat-2-2_20160302.2106, hosted:build_jboss-modules-1-5-0_20160303.0619, hosted:build_20160303092153_kc-twitter_20160303.0922, hosted:build_20160303092153_kc-twitter_20160303.0924, hosted:build_20160303092153_kc-twitter_20160303.0925, hosted:build_jboss-modules-1-5-0_20160311.0431, hosted:build_jboss-modules-1-5-0_20160311.0434, hosted:build_jboss-modules-1-5-0-group-member_20160311.0520, hosted:build_jboss-modules-1-5-0_20160311.1521, hosted:build_jboss-servlet-spec-api-1-0-1_20160311.1525, hosted:build_org-liquibase-liquibase-parent-3-4-1_20160315.0805, hosted:build_com-google-zxing-zxing-parent-3-2-1_20160315.0807, hosted:Freemarker, hosted:build_com-google-zxing-zxing-parent-3-2-1_20160315.1355], getName()=builds-untested, getKey()=group:builds-untested]
2016-03-16 14:41:34.326 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_keycloak-1-9-0-CR1-redhat-2-2_20160302.2106, storage-directory=null]
  HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160303.0619, storage-directory=null]
  HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_20160303092153_kc-twitter_20160303.0922, storage-directory=null]
  HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_20160303092153_kc-twitter_20160303.0924, storage-directory=null]
  HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_20160303092153_kc-twitter_20160303.0925, storage-directory=null]
  HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160311.0431, storage-directory=null]
  HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160311.0434, storage-directory=null]
  HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0-group-member_20160311.0520, storage-directory=null]
  HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160311.1521, storage-directory=null]
  HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-servlet-spec-api-1-0-1_20160311.1525, storage-directory=null]
  HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_org-liquibase-liquibase-parent-3-4-1_20160315.0805, storage-directory=null]
  HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_com-google-zxing-zxing-parent-3-2-1_20160315.0807, storage-directory=null]
  HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:Freemarker, storage-directory=null]
  HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_com-google-zxing-zxing-parent-3-2-1_20160315.1355, storage-directory=null] is a group. Attempting downloads from (in order):
  {}
2016-03-16 14:41:34.326 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.m.g.internal.TransferManagerImpl - Attempting to batch-retrieve 14 resources:
  Resource [location=Cache-only location [hosted:Freemarker], path=org/twitter4j/twitter4j/maven-metadata.xml]
  Resource [location=Cache-only location [hosted:build_jboss-modules-1-5-0_20160311.1521], path=org/twitter4j/twitter4j/maven-metadata.xml]
  Resource [location=Cache-only location [hosted:build_org-liquibase-liquibase-parent-3-4-1_20160315.0805], path=org/twitter4j/twitter4j/maven-metadata.xml]
  Resource [location=Cache-only location [hosted:build_jboss-modules-1-5-0_20160311.0431], path=org/twitter4j/twitter4j/maven-metadata.xml]
  Resource [location=Cache-only location [hosted:build_jboss-modules-1-5-0_20160311.0434], path=org/twitter4j/twitter4j/maven-metadata.xml]
  Resource [location=Cache-only location [hosted:build_com-google-zxing-zxing-parent-3-2-1_20160315.0807], path=org/twitter4j/twitter4j/maven-metadata.xml]
  Resource [location=Cache-only location [hosted:build_com-google-zxing-zxing-parent-3-2-1_20160315.1355], path=org/twitter4j/twitter4j/maven-metadata.xml]
  Resource [location=Cache-only location [hosted:build_jboss-modules-1-5-0-group-member_20160311.0520], path=org/twitter4j/twitter4j/maven-metadata.xml]
  Resource [location=Cache-only location [hosted:build_keycloak-1-9-0-CR1-redhat-2-2_20160302.2106], path=org/twitter4j/twitter4j/maven-metadata.xml]
  Resource [location=Cache-only location [hosted:build_jboss-modules-1-5-0_20160303.0619], path=org/twitter4j/twitter4j/maven-metadata.xml]
  Resource [location=Cache-only location [hosted:build_jboss-servlet-spec-api-1-0-1_20160311.1525], path=org/twitter4j/twitter4j/maven-metadata.xml]
  Resource [location=Cache-only location [hosted:build_20160303092153_kc-twitter_20160303.0922], path=org/twitter4j/twitter4j/maven-metadata.xml]
  Resource [location=Cache-only location [hosted:build_20160303092153_kc-twitter_20160303.0925], path=org/twitter4j/twitter4j/maven-metadata.xml]
  Resource [location=Cache-only location [hosted:build_20160303092153_kc-twitter_20160303.0924], path=org/twitter4j/twitter4j/maven-metadata.xml]
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.DefaultContentManager - Attempting to retrieve: org/twitter4j/twitter4j/maven-metadata.xml from: hosted:build_keycloak-1-9-0-CR1-redhat-2-2_20160302.2106
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_keycloak-1-9-0-CR1-redhat-2-2_20160302.2106, storage-directory=null] via: org.commonjava.indy.core.content.MavenMetadataGenerator@518ce9d9
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.MavenMetadataGenerator - Generating maven-metadata.xml for releases
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_keycloak-1-9-0-CR1-redhat-2-2_20160302.2106, storage-directory=null] via: org.commonjava.indy.core.content.ArchetypeCatalogGenerator@8e9295e
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_keycloak-1-9-0-CR1-redhat-2-2_20160302.2106, storage-directory=null] via: org.commonjava.indy.core.content.HttpMetadataCleanupGenerator@392b2786
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.DefaultContentManager - Attempting to retrieve: org/twitter4j/twitter4j/maven-metadata.xml from: hosted:build_jboss-modules-1-5-0_20160303.0619
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160303.0619, storage-directory=null] via: org.commonjava.indy.core.content.MavenMetadataGenerator@518ce9d9
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.MavenMetadataGenerator - Generating maven-metadata.xml for releases
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160303.0619, storage-directory=null] via: org.commonjava.indy.core.content.ArchetypeCatalogGenerator@8e9295e
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160303.0619, storage-directory=null] via: org.commonjava.indy.core.content.HttpMetadataCleanupGenerator@392b2786
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.DefaultContentManager - Attempting to retrieve: org/twitter4j/twitter4j/maven-metadata.xml from: hosted:build_20160303092153_kc-twitter_20160303.0922
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_20160303092153_kc-twitter_20160303.0922, storage-directory=null] via: org.commonjava.indy.core.content.MavenMetadataGenerator@518ce9d9
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.MavenMetadataGenerator - Generating maven-metadata.xml for releases
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_20160303092153_kc-twitter_20160303.0922, storage-directory=null] via: org.commonjava.indy.core.content.ArchetypeCatalogGenerator@8e9295e
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_20160303092153_kc-twitter_20160303.0922, storage-directory=null] via: org.commonjava.indy.core.content.HttpMetadataCleanupGenerator@392b2786
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.DefaultContentManager - Attempting to retrieve: org/twitter4j/twitter4j/maven-metadata.xml from: hosted:build_20160303092153_kc-twitter_20160303.0924
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_20160303092153_kc-twitter_20160303.0924, storage-directory=null] via: org.commonjava.indy.core.content.MavenMetadataGenerator@518ce9d92016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.MavenMetadataGenerator - Generating maven-metadata.xml for releases
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_20160303092153_kc-twitter_20160303.0924, storage-directory=null] via: org.commonjava.indy.core.content.ArchetypeCatalogGenerator@8e9295e
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_20160303092153_kc-twitter_20160303.0924, storage-directory=null] via: org.commonjava.indy.core.content.HttpMetadataCleanupGenerator@392b2786
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.DefaultContentManager - Attempting to retrieve: org/twitter4j/twitter4j/maven-metadata.xml from: hosted:build_20160303092153_kc-twitter_20160303.0925
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_20160303092153_kc-twitter_20160303.0925, storage-directory=null] via: org.commonjava.indy.core.content.MavenMetadataGenerator@518ce9d9
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.MavenMetadataGenerator - Generating maven-metadata.xml for releases
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_20160303092153_kc-twitter_20160303.0925, storage-directory=null] via: org.commonjava.indy.core.content.ArchetypeCatalogGenerator@8e9295e
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_20160303092153_kc-twitter_20160303.0925, storage-directory=null] via: org.commonjava.indy.core.content.HttpMetadataCleanupGenerator@392b2786
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.DefaultContentManager - Attempting to retrieve: org/twitter4j/twitter4j/maven-metadata.xml from: hosted:build_jboss-modules-1-5-0_20160311.0431
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160311.0431, storage-directory=null] via: org.commonjava.indy.core.content.MavenMetadataGenerator@518ce9d9
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.MavenMetadataGenerator - Generating maven-metadata.xml for releases
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160311.0431, storage-directory=null] via: org.commonjava.indy.core.content.ArchetypeCatalogGenerator@8e9295e
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160311.0431, storage-directory=null] via: org.commonjava.indy.core.content.HttpMetadataCleanupGenerator@392b2786
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.DefaultContentManager - Attempting to retrieve: org/twitter4j/twitter4j/maven-metadata.xml from: hosted:build_jboss-modules-1-5-0_20160311.0434
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160311.0434, storage-directory=null] via: org.commonjava.indy.core.content.MavenMetadataGenerator@518ce9d9
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.MavenMetadataGenerator - Generating maven-metadata.xml for releases
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160311.0434, storage-directory=null] via: org.commonjava.indy.core.content.ArchetypeCatalogGenerator@8e9295e
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160311.0434, storage-directory=null] via: org.commonjava.indy.core.content.HttpMetadataCleanupGenerator@392b2786
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.DefaultContentManager - Attempting to retrieve: org/twitter4j/twitter4j/maven-metadata.xml from: hosted:build_jboss-modules-1-5-0-group-member_20160311.0520
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0-group-member_20160311.0520, storage-directory=null] via: org.commonjava.indy.core.content.MavenMetadataGenerator@518ce9d9
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.MavenMetadataGenerator - Generating maven-metadata.xml for releases
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0-group-member_20160311.0520, storage-directory=null] via: org.commonjava.indy.core.content.ArchetypeCatalogGenerator@8e9295e
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0-group-member_20160311.0520, storage-directory=null] via: org.commonjava.indy.core.content.HttpMetadataCleanupGenerator@392b2786
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.DefaultContentManager - Attempting to retrieve: org/twitter4j/twitter4j/maven-metadata.xml from: hosted:build_jboss-modules-1-5-0_20160311.1521
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160311.1521, storage-directory=null] via: org.commonjava.indy.core.content.MavenMetadataGenerator@518ce9d9
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.MavenMetadataGenerator - Generating maven-metadata.xml for releases
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160311.1521, storage-directory=null] via: org.commonjava.indy.core.content.ArchetypeCatalogGenerator@8e9295e
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-modules-1-5-0_20160311.1521, storage-directory=null] via: org.commonjava.indy.core.content.HttpMetadataCleanupGenerator@392b2786
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.DefaultContentManager - Attempting to retrieve: org/twitter4j/twitter4j/maven-metadata.xml from: hosted:build_jboss-servlet-spec-api-1-0-1_20160311.1525
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-servlet-spec-api-1-0-1_20160311.1525, storage-directory=null] via: org.commonjava.indy.core.content.MavenMetadataGenerator@518ce9d9
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.MavenMetadataGenerator - Generating maven-metadata.xml for releases
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-servlet-spec-api-1-0-1_20160311.1525, storage-directory=null] via: org.commonjava.indy.core.content.ArchetypeCatalogGenerator@8e9295e
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_jboss-servlet-spec-api-1-0-1_20160311.1525, storage-directory=null] via: org.commonjava.indy.core.content.HttpMetadataCleanupGenerator@392b2786
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.DefaultContentManager - Attempting to retrieve: org/twitter4j/twitter4j/maven-metadata.xml from: hosted:build_org-liquibase-liquibase-parent-3-4-1_20160315.0805
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_org-liquibase-liquibase-parent-3-4-1_20160315.0805, storage-directory=null] via: org.commonjava.indy.core.content.MavenMetadataGenerator@518ce9d9
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.MavenMetadataGenerator - Generating maven-metadata.xml for releases
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_org-liquibase-liquibase-parent-3-4-1_20160315.0805, storage-directory=null] via: org.commonjava.indy.core.content.ArchetypeCatalogGenerator@8e9295e
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_org-liquibase-liquibase-parent-3-4-1_20160315.0805, storage-directory=null] via: org.commonjava.indy.core.content.HttpMetadataCleanupGenerator@392b2786
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.DefaultContentManager - Attempting to retrieve: org/twitter4j/twitter4j/maven-metadata.xml from: hosted:build_com-google-zxing-zxing-parent-3-2-1_20160315.0807
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_com-google-zxing-zxing-parent-3-2-1_20160315.0807, storage-directory=null] via: org.commonjava.indy.core.content.MavenMetadataGenerator@518ce9d9
2016-03-16 14:41:34.327 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.MavenMetadataGenerator - Generating maven-metadata.xml for releases
2016-03-16 14:41:34.328 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_com-google-zxing-zxing-parent-3-2-1_20160315.0807, storage-directory=null] via: org.commonjava.indy.core.content.ArchetypeCatalogGenerator@8e9295e
2016-03-16 14:41:34.328 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:build_com-google-zxing-zxing-parent-3-2-1_20160315.0807, storage-directory=null] via: org.commonjava.indy.core.content.HttpMetadataCleanupGenerator@392b2786
2016-03-16 14:41:34.328 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.DefaultContentManager - Attempting to retrieve: org/twitter4j/twitter4j/maven-metadata.xml from: hosted:Freemarker
2016-03-16 14:41:34.328 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] DEBUG o.c.i.c.c.DefaultContentManager - Attempting to generate content for path: org/twitter4j/twitter4j/maven-metadata.xml in: HostedRepository [allowSnapshots=true, allowReleases=true, key=hosted:Freemarker, storage-directory=null] via: org.commonjava.indy.core.content.MavenMetadataGenerator@518ce9d9
2016-03-16 14:41:34.328 [GET /group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml] INFO  o.c.i.c.c.MavenMetadataGenerator - Generating maven-metadata.xml for releases
2016-03-16 14:41:34.329 [XNIO-2 task-14] ERROR io.undertow.request - UT005023: Exception handling request to /api/group/builds-untested/org/twitter4j/twitter4j/maven-metadata.xml
org.jboss.resteasy.spi.UnhandledException: java.lang.NullPointerException
        at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:76) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
        at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:212) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
        at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:149) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:372) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:179) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
        at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:220) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
        at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
        at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) ~[jboss-servlet-api_3.1_spec-1.0.0.Final.jar:1.0.0.Final]
        at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:130) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at org.commonjava.indy.bind.jaxrs.ResourceManagementFilter.doFilter(ResourceManagementFilter.java:60) ~[indy-embedder-savant-1.0-SNAPSHOT.jar:na]
        at org.commonjava.indy.bind.jaxrs.ResourceManagementFilter$Proxy$_$$_WeldClientProxy.doFilter(Unknown Source) ~[indy-embedder-savant-1.0-SNAPSHOT.jar:na]
        at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:60) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:132) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:85) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:61) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:56) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:45) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:63) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:58) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:70) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.security.handlers.SecurityInitialHandler.handleRequest(SecurityInitialHandler.java:76) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
        at org.commonjava.indy.bind.jaxrs.HeaderDebugger.handleRequest(HeaderDebugger.java:93) ~[indy-embedder-savant-1.0-SNAPSHOT.jar:na]
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:261) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:247) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:76) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:166) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.server.Connectors.executeRootHandler(Connectors.java:197) [undertow-core-1.1.2.Final.jar:1.1.2.Final]
        at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:764) [undertow-core-1.1.2.Final.jar:1.1.2.Final]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_71]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_71]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_71]
Caused by: java.lang.NullPointerException: null
        at org.commonjava.indy.core.content.MavenMetadataGenerator.writeVersionMetadata(MavenMetadataGenerator.java:395) ~[indy-embedder-savant-1.0-SNAPSHOT.jar:na]
        at org.commonjava.indy.core.content.MavenMetadataGenerator.generateFileContent(MavenMetadataGenerator.java:196) ~[indy-embedder-savant-1.0-SNAPSHOT.jar:na]
        at org.commonjava.indy.core.content.DefaultContentManager.doRetrieve(DefaultContentManager.java:294) ~[indy-embedder-savant-1.0-SNAPSHOT.jar:na]
        at org.commonjava.indy.core.content.DefaultContentManager.retrieve(DefaultContentManager.java:256) ~[indy-embedder-savant-1.0-SNAPSHOT.jar:na]
        at org.commonjava.indy.core.ctl.ContentController.get(ContentController.java:147) ~[indy-embedder-savant-1.0-SNAPSHOT.jar:na]
        at org.commonjava.indy.core.ctl.ContentController.get(ContentController.java:140) ~[indy-embedder-savant-1.0-SNAPSHOT.jar:na]
        at org.commonjava.indy.core.ctl.ContentController$Proxy$_$$_WeldClientProxy.get(Unknown Source) ~[indy-embedder-savant-1.0-SNAPSHOT.jar:na]
        at org.commonjava.indy.core.bind.jaxrs.ContentAccessHandler.doGet(ContentAccessHandler.java:294) ~[indy-embedder-savant-1.0-SNAPSHOT.jar:na]
        at org.commonjava.indy.core.bind.jaxrs.ContentAccessHandler$Proxy$_$$_WeldClientProxy.doGet(Unknown Source) ~[indy-embedder-savant-1.0-SNAPSHOT.jar:na]
        at sun.reflect.GeneratedMethodAccessor172.invoke(Unknown Source) ~[na:na]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_71]
        at java.lang.reflect.Method.invoke(Method.java:497) ~[na:1.8.0_71]
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:137) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:296) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:250) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:237) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:356) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
        ... 34 common frames omitted
```

--- a/core/src/main/java/org/commonjava/indy/core/content/MavenMetadataGenerator.java
+++ b/core/src/main/java/org/commonjava/indy/core/content/MavenMetadataGenerator.java
@@ -351,7 +351,7 @@ public class MavenMetadataGenerator
     {
         ArtifactPathInfo samplePomInfo = null;
 
-        // first level will contain version directories...for each directory, we need to verify the presence of a .pom file before including 
+        // first level will contain version directories...for each directory, we need to verify the presence of a .pom file before including
         // as a valid version
         final List<SingleVersion> versions = new ArrayList<SingleVersion>();
         nextTopResource: for ( final StoreResource topResource : firstLevelFiles )
@@ -365,13 +365,18 @@ public class MavenMetadataGenerator
                     if ( fileResource.getPath()
                                      .endsWith( ".pom" ) )
                     {
-                        versions.add( VersionUtils.createSingleVersion( new File( topPath ).getName() ) );
-                        if ( samplePomInfo == null )
+                        ArtifactPathInfo filePomInfo = ArtifactPathInfo.parse( fileResource.getPath() );
+                        // check if the pom is valid for the path
+                        if ( filePomInfo != null )
                         {
-                            samplePomInfo = ArtifactPathInfo.parse( fileResource.getPath() );
-                        }
+                            versions.add( VersionUtils.createSingleVersion( new File( topPath ).getName() ) );
+                            if ( samplePomInfo == null )
+                            {
+                                samplePomInfo = filePomInfo;
+                            }
 
-                        continue nextTopResource;
+                            continue nextTopResource;
+                        }
                     }
                 }
             }

